org.jacoco.core/src/org/jacoco/core/internal/analysis/ClassAnalyzer.java

```diff
@@ -82,8 +83,24 @@ public void visitEnd() {
 		};
 	}
 
+	/**
+	 * @return <code>true</code> if method should not be analyzed
+	 */
 	// TODO: Use filter hook in future
-	private boolean isMethodFiltered(final int access, final String name) {
+	private boolean isMethodFiltered(final String className,
+			final String superClassName, final int access, final String name,
+			final String desc) {
+		if ("java/lang/Enum".equals(superClassName)) {
+			// filter out methods that compiler creates for enums
+			if ("values".equals(name)
```

Comment:
We mix two different comparisons styles here: "expected".equals(actual) vs. actual.equals("expected"). If would prefer to use the first one only as it is NPE safe.